task04.rb

```
@@ -0,0 +1,38 @@
+#For a given array of elements show the type of each element but keep the array structure, e.g. [["hi"], {}, :sym, [1, true], nil] should output [[String], Hash, Symbol, [Integer, TrueClass], NilClass]
+
+def task04(arr)
+	result_array = []
+
+	def check_type(el, result_array)
+		if el.is_a?(Array)
+			if el.empty?
+				Array
```

@MaraEfimova Hm, didn't we talk that in case of empty array the method should throw an error (e.g. if the method is called like `task04([])`)? In addition - I believe that if you pass a hash/range/etc. instead of an array - the method should also throw an error because this is considered as invalid input.